### PR TITLE
use api repo constants

### PR DESCRIPTION
**Pull request by xichengliudui**
**Please provide a description of this PR:**

use api repo constants

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [x] Configuration Infrastructure

### CODE REVIEW

**Comment by xichengliudui** [comment]
```
[root@iZt4n97e16shwk34ivinm5Z istio]# make lint
useradd: UID 0 is not unique
INFO [config_reader] Used config file common/config/.golangci.yml 
INFO [lintersdb] Active 19 linters: [deadcode errcheck exportloopref gocritic gofumpt goimports golint gosimple govet ineffassign lll misspell staticcheck structcheck stylecheck typecheck unconvert unparam varcheck] 
INFO [loader] Using build tags: [integ integfuzz] 
INFO [loader] Go packages loading at mode 575 (imports|name|exports_file|files|types_sizes|compiled_files|deps) took 22.953950229s 
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive. 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 357.686076ms 
common/scripts/lint_go.sh: line 28:  3349 Killed                  golangci-lint run -v -c ./common/config/.golangci.yml --out-format colored-line-number,junit-xml:"${ARTIFACTS}"/junit-lint.xml
make[1]: *** [common/Makefile.common.mk:46: lint-go] Error 123
make: *** [lint] Error 2
```

File: pilot/pkg/xds/mesh_network_test.go [hunk 1]
```diff
@@ -227,7 +227,7 @@ func TestMeshNetworking(t *testing.T) {
 						Labels: map[string]string{
 							label.TopologyNetwork.Name: "network-2",
 							// set the label here to test it = expectation doesn't change since we map back to that via NodePort
-							controller.IstioGatewayPortLabel: "443",
+							annotation.NetworkingGatewayPort.Name: "443",
```
Comment (howardjohn) [inline pilot/pkg/xds/mesh_network_test.go:230]: why is it "annotation" if its in a label? can be it be label or annotation?

Comment (xichengliudui) [reply pilot/pkg/xds/mesh_network_test.go:230]: `IstioGatewayPortLabel` is moved into the `annotation` package
https://github.com/istio/api/blob/5796da5cb825caed340bab1c7eacd8290b625f35/annotation/annotations.gen.go#L269

Comment (xichengliudui) [reply pilot/pkg/xds/mesh_network_test.go:230]: do I need to move `IstioGatewayPortLabel` to the `label` package? 
i can do it

**Pull request merged**